docs/getting-started.md

```diff
@@ -0,0 +1,36 @@
+# Getting started with developing your own Custom Metric Server API
+This is an example of how to vendor this project and setup your own provider.  This example uses [glide](https://github.com/Masterminds/glide) but you can use any other [dependency management tool](https://github.com/golang/go/wiki/PackageManagementTools) of your choice.
+
+## Create your new project:
+
+- `mkdir $GOPATH/src/github.com/your-company/my-custom-metric-server`
+- `cd $GOPATH/src/github.com/your-company/my-custom-metric-server`
+
+## Vendor this repository:
+
+- `glide create`
+- `glide get github.com/kubernetes-incubator/custom-metrics-apiserver`
+
+## Create the entry point from the sample:
+
+- `touch main.go` 
+
+> See the [sample entry point](https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/master/sample-main.go) for a complete example
+
+## Add the server:
+
+- `mkdir -p pkg/cmd/server`
+- `touch pkg/cmd/server/start.go` 
+
+> See the [sample server](https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/master/pkg/sample-cmd/server/start.go) for complete example
+
+## Add and implement [the custom metrics interface](https://github.com/kubernetes-incubator/custom-metrics-apiserver/blob/d8f23423aa1d0ff2bc9656da863d721725b3c68a/pkg/provider/interfaces.go#L84) for your provider:
+
+- `mkdir -p pkg/cmd/provider`
+- `touch pkg/cmd/server/provider.go` 
```

Comment:
just put this under `pkg/provider`.

In general: 
- command-related stuff goes under `cmd`
- actual functionality goes under `pkg`

Reply:
see comment above about structure